pkg/apis/extensions/v1alpha1/types_defaults.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ErrorCode is a string alias.
+type ErrorCode string
+
+const (
+	// ErrorInfraUnauthorized indicates that the last error occurred due to invalid cloud provider credentials.
+	ErrorInfraUnauthorized ErrorCode = "ERR_INFRA_UNAUTHORIZED"
```

Comment:
These codes are very limited when it comes to extensibility - we might have to weaken the guarantees we have here / remove these constants and go with plain strings.